workbench-core/authorization/src/dynamicAuthorization/ddbDynamicAuthorizationPermissionsPlugin.ts

```diff
@@ -60,7 +62,30 @@ export class DDBDynamicAuthorizationPermissionsPlugin implements DynamicAuthoriz
   public async getIdentityPermissionsByIdentity(
     getIdentityPermissionsByIdentityRequest: GetIdentityPermissionsByIdentityRequest
   ): Promise<GetIdentityPermissionsByIdentityResponse> {
-    throw new Error('Method not implemented.');
+    const { identityId, identityType } = getIdentityPermissionsByIdentityRequest;
+    const identity = `${identityType}|${identityId}`;
+    const queryParams = {
+      index: this._getIdentityPermissionsByIdentityIndex,
+      key: {
+        name: this._getIdentityPermissionsByIdentityPartitionKey,
+        value: identity
+      }
+    };
+    let { data, paginationToken } = await this._dynamoDBService.getPaginatedItems(queryParams);
+
+    while (paginationToken) {
```

Comment:
This isn't something I'm super concerned about but want to get on your radar: This is a potential DoS issue, but only at scale. If someone accidentally created enough permissions, they could time out the call by making this loop run for longer than the response timeout. There is possibly also a response-body-size related strangling, but with serverless architecture that may be less of an issue. Anyway, not something I'm worried about for now, but if there is a way to return this data in a paginated way, we could avoid the issue completely.

Reply:
Yeah the DoS attack is interesting, these APIs should not be exposed at the handler level directly anyway.

Reply:
API exposure isn't the problem, its the amount of data being retrieved. We're ok for small sets of data, but we'd e ok for big sets of data too if we could return just the page of results instead of the entire aggregation of all results.

Reply:
added paginated results